### ISSUE TITLE
GNSS position update delay

### ISSUE THREAD

**Issue by MichaelusSchaperus** [zauberzeug/field_friend#38]
Hello,

we have issues concerning the GNSS position update. 

Problem description:
_**tl;dr**: The GNSS position takes up to a few (1-5) minutes to get updated so the robot is not even close to where he should be._

After planning a flield with two rows, we want the robot to simply follow them using the weeding automation in monitoring only mode. The rows doesn't exist in real life, they are are only defined per GPS locations, so there are no plants to detect or something like that. After starting the automation the robot moves towards the row. We can see the robot moving inside the simulation in the UI, cause it's proably using the prediction from the Odometer. The real robot however is not on the depicted position, cause it's not getting new GNSS information. It takes up to a few minutes to get updated and we get the following warning: 
```
[WARNING] field_friend/navigation/gnss.py:236: GNSS distance to prediction to high: 35.18m!!
```

We get a similar problem when trying to add a new field using the field planner page. We tried to use the robot to add the corner points of the field. After finishing the field planning, we saw that all points where exactly the same.
So the problem is not based on an automation.

It seems like the position update is buffered and gets released with delay.

What we also checked so far:
- RTK fix is given at all times
- the GNSS device (Septentrio) is updating and publishing the correct position in real time

**Comment by angelom93** [comment]
Hello @MichaelusSchaperus,

The problem was caused by an outdatet RoSys version running on your system. We fixed a problem with the arp_scan method in version [v0.10.5](https://github.com/zauberzeug/rosys/releases/tag/v0.10.5). This caused a blocking of I/O-bound methods, specifically the serial read method in your case.

Thank you for pointing out the problem.
